pkg/virt-operator/controller/components/crds.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+package components
+
+import (
+	"fmt"
+
+	extv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	extclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+)
+
+func CreateCRDs(clientset kubecli.KubevirtClient) error {
+
+	ext, err := extclient.NewForConfig(clientset.Config())
+	if err != nil {
+		return fmt.Errorf("unable to create apiextensions client: %v", err)
+	}
+
+	objects := []*extv1beta1.CustomResourceDefinition{
+		NewVirtualMachineInstanceCrd(),
+		NewVirtualMachineCrd(),
+		NewReplicaSetCrd(),
+		NewPresetCrd(),
+		NewVirtualMachineInstanceMigrationCrd(),
+	}
+
+	for _, crd := range objects {
+		_, err := ext.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return fmt.Errorf("unable to create crd %+v: %v", crd, err)
+		}
+	}
+
+	return nil
+}
+
+func NewKubeVirtCRD(clientset kubecli.KubevirtClient) error {
```

Comment:
very good question. I wasn't sure yet if the Operator should install the KubeVirt CRD itself, or if it will be installed by manifest alongside with the Operator deployment manifest, or later by OLM... so, for the moment I decided to leave it in the manifest